lib/data/wrapper.js

```diff
@@ -26,6 +26,22 @@ if (config.backends.data === 'mem') {
     implName = 'sproxyd';
 }
 
+/**
+ * _retryDelete - Attempt to delete key again if it failed previously
+ * @param  {string} key - location of the object to delete
+ * @param  {object} log - Werelogs request logger
+ * @returns {boolean} true or false - indicates whether delete was successful
+ */
+function _retryDelete(key, log) {
```

Comment:
Is this function working as intended ?
For what I see and understand, it's always returning ```undefined```.
I think you wanted to write something more like:

```es6
function _retryDelete(key, log) {
    log.trace('retrying delete one more time...');
    return client.delete(key, log.getSerializedUids(), err => {
        if (err) {
            return false;
        }
        return true;
    });
}
```

But this form only works when the call to ```client.delete()``` is synchronous, if it is asynchronous, as with sprocydclient I guess, you'll still always undefined.

Maybe what you want is more like
```es6
const MAX_RETRY = 2;
function _retryDelete(key, log, count, cb) {
    if (count > MAX_RETRY) {
        return cb(errors.InternalError);
    }
    log.trace('retrying delete one more time...');
    return client.delete(key, log.getSerializedUids(), err => {
        if (err) {
            return _retryDelete(key, log, ++count, cb);
        }
        return cb();
    });
}
```

And the delete function become:

```es6
delete: (objectGetInfo, log, cb) => {
    const callback = cb || log.end;
    // If objectGetInfo.key exists the md-model-version is 2 or greater.
    // Otherwise, the objectGetInfo is just the key string.
    const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;
    log.debug('sending delete to datastore', {
        implName, key,
        method: 'delete',
    });
    _retryDelete(key, log, 0, err => {
        if (err) {
            log.error('error deleting object from datastore', {
                error: err ,
                location: key
            });
        }
        return callback(err);
    });
}
```

With this solution you'll try MAX_RETRY times before calling the callback with an error.